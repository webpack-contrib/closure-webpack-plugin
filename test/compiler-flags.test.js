import path from 'path';
import ClosureWebpackPlugin from '../src';
import testCompiler from './helpers/compiler';

describe('Closure: Compiler flags', () => {
  jest.setTimeout(60 * 1000);

  describe('externs', () => {
    test('only path', async () => {
      expect.assertions(1);
      const stats = await testCompiler('externs-flag', {
        plugins: [
          new ClosureWebpackPlugin({ mode: 'STANDARD' }, {
            externs: path.resolve(__dirname, './fixtures/externs-flag/externs.js'),
            compilation_level: 'ADVANCED',
          }),
        ],
      });

      const fs = stats.compilation.compiler.outputFileSystem;
      const pathToBundle = fs.join(__dirname, './results/main.bundle.js');
      const bundleContent = fs.readFileSync(pathToBundle, 'utf8');
      expect(bundleContent).toMatchSnapshot();
    });

    test('array of paths', async () => {
      expect.assertions(1);
      const stats = await testCompiler('externs-2-flag', {
        plugins: [
          new ClosureWebpackPlugin({ mode: 'STANDARD' }, {
            externs: [
              path.resolve(__dirname, './fixtures/externs-2-flag/externs.js'),
              path.resolve(__dirname, './fixtures/externs-2-flag/externs2.js'),
            ],
            compilation_level: 'ADVANCED',
          }),
        ],
      });

      const fs = stats.compilation.compiler.outputFileSystem;
      const pathToBundle = fs.join(__dirname, './results/main.bundle.js');
      const bundleContent = fs.readFileSync(pathToBundle, 'utf8');
      expect(bundleContent).toMatchSnapshot();
    });

    test('wrong value', async () => {
      expect.assertions(1);
      const stats = await testCompiler('externs-2-flag', {
        plugins: [
          new ClosureWebpackPlugin({ mode: 'STANDARD' }, {
            externs: {},
            compilation_level: 'WHITESPACE_ONLY',
          }),
        ],
      });

      const { errors } = stats.compilation;
      const stringifyErrors = errors.map(e => e.toString());
      const matcher = expect.stringMatching(/Cannot read file \[object Object\]:/);
      expect(stringifyErrors).toEqual(
        expect.arrayContaining([matcher]));
    });

    test('array contain wrong value', async () => {
      expect.assertions(1);
      const stats = await testCompiler('externs-2-flag', {
        plugins: [
          new ClosureWebpackPlugin({ mode: 'STANDARD' }, {
            externs: [{}],
            compilation_level: 'WHITESPACE_ONLY',
          }),
        ],
      });

      const { errors } = stats.compilation;
      const stringifyErrors = errors.map(e => e.toString());
      const matcher = expect.stringMatching(/Cannot read file \[object Object\]:/);
      expect(stringifyErrors).toEqual(
        expect.arrayContaining([matcher]));
    });
  });
});
