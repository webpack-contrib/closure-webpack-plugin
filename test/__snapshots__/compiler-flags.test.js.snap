// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Closure: Compiler flags externs array of paths 1`] = `
"'use strict';(function(e){function b(a){if(d[a])return d[a].a;var c=d[a]={m:a,f:!1,a:{}};e[a].call(c.a,c,c.a,b);c.f=!0;return c.a}var d={};b.l=e;b.j=d;b.b=function(a,c){b.c(a)||Object.defineProperty(a,\\"a\\",{configurable:!1,enumerable:!0,get:c})};b.h=function(a){var c=a&&a.g?function(){return a[\\"default\\"]}:function(){return a};b.b(c,c);return c};b.c=function(a){return Object.prototype.hasOwnProperty.call(a,\\"a\\")};b.i=\\"\\";return b(b.o=0)})([function(){test(1);test2(2)}]);

//# sourceMappingURL=main.bundle.js.map"
`;

exports[`Closure: Compiler flags externs only path 1`] = `
"'use strict';(function(e){function b(a){if(d[a])return d[a].a;var c=d[a]={m:a,f:!1,a:{}};e[a].call(c.a,c,c.a,b);c.f=!0;return c.a}var d={};b.l=e;b.j=d;b.b=function(a,c){b.c(a)||Object.defineProperty(a,\\"a\\",{configurable:!1,enumerable:!0,get:c})};b.h=function(a){var c=a&&a.g?function(){return a[\\"default\\"]}:function(){return a};b.b(c,c);return c};b.c=function(a){return Object.prototype.hasOwnProperty.call(a,\\"a\\")};b.i=\\"\\";return b(b.o=0)})([function(){test(1)}]);

//# sourceMappingURL=main.bundle.js.map"
`;
